pkg/tags/tags.go

```diff
@@ -53,17 +53,29 @@ type Tags struct {
 
 // NewTags creates a tags object
 func NewTags(stateStore storage.StateStorer, logger logging.Logger) *Tags {
+	rand.Seed(time.Now().UnixNano())
```

Comment:
this is not so great as it sets the global rand seed. we should generally refrain from doing such things. it would be better to just pass a new source here as a parameter and do the same within the test. it should simplify things and clean this up

Reply:
well, we can use a different source, it needs a lock, as only the global rand is safe to use in concurrency
I will replace it